app/assets/javascripts/analytics.js.erb

```diff
@@ -2,6 +2,28 @@
 var element = document.querySelector('[data-module="cookie-banner"]')
 new GOVUK.Modules.CookieBanner().start($(element))
 
+var referenceNumberPattern = function() {
+  var pattern = /\d{8}-\d{6}-.{6}/
+  return pattern
+}
+
+var productIdPattern = function() {
+  var pattern = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/
+  return pattern
```

Comment:
You could use `var PRODUCT_PATTERN = /\b[0-9a-f]{8}\...` at the top of the file, since this can be treated as a constant. Likewise for the `REF_NUMBER_PATTERN`.

Reply:
👍